### PR TITLE
Update android gradle plugin to 2.3.0 and gradle wrapper to 3.4

### PR DESCRIPTION
**Pull request by zaki50**
android grade plugin 2.3.0 has been released.

This improves dependency resolution time at the beginning of gradle command much!

@realm/java

### CODE REVIEW

**Comment by bmeike** [comment]
Are you sure you wanna do this?  We'll just have to do it again for the actual release...

**Comment by zaki50** [comment]
I don't merge this until actual 2.3.0 will be released.

**Comment by zaki50** [comment]
@cmelchior Yes. `wrapper` task in each `build.gradle` uses `gradleVersion` property in `realm.properties`.

File: examples/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Tue Sep 20 14:25:59 CST 2016
+#Thu Mar 02 03:11:13 JST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (cmelchior) [inline examples/gradle/wrapper/gradle-wrapper.properties]: 3.4 ?

File: realm-annotations/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Tue Sep 20 14:22:29 CST 2016
+#Thu Mar 02 03:11:50 JST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (cmelchior) [inline realm-annotations/gradle/wrapper/gradle-wrapper.properties]: 3.4

File: realm-transformer/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Tue Sep 20 14:33:05 CST 2016
+#Thu Mar 02 03:11:56 JST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (cmelchior) [inline realm-transformer/gradle/wrapper/gradle-wrapper.properties]: 3.4

**Comment by zaki50** [comment]
@cmelchior Some modules didn't respect `realm.properties`. FIxed.

**Comment by zaki50** [comment]
I've cleaned up commit history.

**Pull request merged**